poc/tenant-controller/pkg/controllers/tenants/controller.go

```diff
@@ -63,7 +70,21 @@ func (c *Controller) Run(ctx context.Context) error {
 func (c *Controller) createTenant(obj *tenantsapi.Tenant) {
 	glog.Info("createTenant: %#v", obj)
 	syncRBACForTenantCRD(obj)
-	// TODO create namespace
+
+	// TODO Add later ... sanity checks to ensure namespaces being requested are valid and not already assigned to another tenant
+	c.tenantsList = make(map[string]tenantsapi.Tenant)
+	c.tenantsList[obj.ObjectMeta.Name] = *obj
+
+	namespace := corev1.Namespace{}
+
+	for n, _ := range obj.Spec.Namespaces {
+		namespace.ObjectMeta.Name = obj.Spec.Namespaces[n].Name
```

Comment:
It's better to construct a namespace object inline, as there's not too much info to fill in. like:

c.k8sclient.CoreV1().Namespaces().Create(&corev1.Namespace{...})

Reply:
It seems to be a style issue mostly ... but okay I can change it to be inline as suggested.  One minor benefit of the current code is that we do not need to allocate the object during every loop. Anyway will change it as suggested.